metadata-ingestion/tests/integration/powerbi_report_server/test_powerbi_report_server.py

```diff
@@ -0,0 +1,117 @@
+from datetime import datetime
+from unittest import mock
+from unittest.mock import patch
+
+from freezegun import freeze_time
+
+from datahub.ingestion.run.pipeline import Pipeline
+from datahub.ingestion.source.powerbi_report_server.report_server_domain import (
+    PowerBiReport,
+    Report,
+)
+from tests.test_helpers import mce_helpers
+
+FROZEN_TIME = "2022-07-08 07:00:00"
+
+POWERBI_REPORT_SERVER_API = "datahub.ingestion.source.powerbi_report_server.report_server.PowerBiReportServerAPI"
+POWERBI_REPORT_SERVER_USER_DAO_GET_OWNER = "datahub.ingestion.source.powerbi_report_server.report_server.UserDao.get_owner_by_name"
+
+
+@freeze_time(FROZEN_TIME)
+@patch(POWERBI_REPORT_SERVER_USER_DAO_GET_OWNER)
+def test_powerbi_report_server_ingest(pytestconfig, tmp_path, mock_time):
```

Comment:
It would be good if we mock the request library to cover most of the business logic in the integration test case.

Please refer power-bi source for the integration test: 
https://github.com/mohdsiddique/datahub/blob/master%2Bpowerbi-optional-user-ingestion/metadata-ingestion/tests/integration/powerbi/test_powerbi.py

Reply:
changed golden file, test case cover most of business logic (getting report metadata and ingest)

Reply:
Please check https://github.com/mohdsiddique/datahub/blob/master%2Bpowerbi-optional-user-ingestion/metadata-ingestion/tests/integration/powerbi/test_powerbi.py and implement the request mock.

the golden file is a validation of ingestion output.

Reply:
written tests according to your example

Reply:
> Please check https://github.com/mohdsiddique/datahub/blob/master%2Bpowerbi-optional-user-ingestion/metadata-ingestion/tests/integration/powerbi/test_powerbi.py and implement the request mock.
> 
> the golden file is a validation of ingestion output.

rewritten tests according to your example